packages/headless/src/utils/validate-payload.ts

```diff
@@ -1,21 +1,42 @@
-import {SchemaDefinition, SchemaValue, Schema} from '@coveo/bueno';
+import {
+  SchemaDefinition,
+  SchemaValue,
+  Schema,
+  SchemaValidationError,
+} from '@coveo/bueno';
 
 export const validatePayloadSchema = <P>(
   payload: P,
-  schemaDefinition: SchemaDefinition<Required<P>>
+  schemaDefinition: SchemaDefinition<Required<P>>,
+  shouldThrowWhenInvalidated = false
 ) => {
   const schema = new Schema(schemaDefinition);
-  const validatedPayload = schema.validate(payload);
-  return {payload: validatedPayload as P};
+  try {
+    const validatedPayload = schema.validate(payload);
+    return {payload: validatedPayload as P};
+  } catch (error) {
+    if (shouldThrowWhenInvalidated) {
+      throw error;
+    }
+    return {payload, error: error as SchemaValidationError};
+  }
 };
 
 export const validatePayloadValue = <P>(
   payload: P,
-  schemaValue: SchemaValue<P>
+  schemaValue: SchemaValue<P>,
+  shouldThrowWhenInvalidated = false
```

Comment:
Why would the parameter be `false` sometimes? Don't we always want to throw when the payload is invalid?